src/extensions/struct-conn/index.ts

```diff
@@ -0,0 +1,365 @@
+}
+
+/** Create visuals for residues and atoms involved in a struct_conn and zoom on them */
+async function addStructConnInspection(plugin: PluginContext, structNode: StructNode, conn: StructConnRecord) {
+    if (!structNode) return;
+
+    const expressionByResidues = structConnExpression(conn, 'residues');
+    const expressionByAtoms = structConnExpression(conn, 'atoms');
+
+    const selectionByResidues = await plugin.build().to(structNode).apply(
+        StructureComponent,
+        { label: `${conn.id} (residues)`, type: { name: 'expression', params: expressionByResidues } },
+        { tags: [TAGS.RESIDUE_SEL] }
+    ).commit();
+    const selectionByAtom = await plugin.build().to(structNode).apply(
+        StructureComponent,
+        { label: `${conn.id} (atoms)`, type: { name: 'expression', params: expressionByAtoms } },
+        { tags: [TAGS.ATOM_SEL] }
+    ).commit();
+
+    if (selectionByResidues.isOk) {
+        await plugin.build().to(selectionByResidues).apply(
+            StructureRepresentation3D,
+            RESIDUES_VISUAL_PARAMS,
+            { tags: [TAGS.RESIDUE_REPR] }
+        ).commit();
+    }
+    if (selectionByAtom.isOk) {
+        const atomBalls = await plugin.build().to(selectionByAtom).apply(
+            StructureRepresentation3D,
+            ATOMS_VISUAL_PARAMS,
+            { tags: [TAGS.ATOM_REPR] }
+        ).commit();
+        plugin.managers.camera.focusRenderObjects(atomBalls.data?.repr.renderObjects, { extraRadius: EXTRA_RADIUS });
+    }
+}
+
+/** Remove anything created by `addStructConnInspection` */
+async function removeAllStructConnInspections(plugin: PluginContext) {
+    const selNodes = [
+        ...plugin.state.data.selectQ(q => q.root.subtree().withTag(TAGS.RESIDUE_SEL)),
+        ...plugin.state.data.selectQ(q => q.root.subtree().withTag(TAGS.ATOM_SEL)),
+    ];
+    for (const node of selNodes) {
+        await plugin.build().delete(node).commit();
+    }
+}
+
+/** Hide all carbohydrate SNFG visuals */
+function hideSnfgNodes(plugin: PluginContext) {
+    const hiddenNodes = getExtensionState(plugin).hiddenNodes;
+    const snfgNodes = plugin.state.data.selectQ(q => q.root.subtree().withTag('branched-snfg-3d'));
+    for (const node of snfgNodes) {
+        if (!node.state.isHidden) {
+            setSubtreeVisibility(plugin.state.data, node.transform.ref, true); // true means hidden
+            hiddenNodes.set(node.transform.ref, true);
+        }
+    }
+}
+
+/** Make visible all carbohydrate SNFG visuals that have been hidden by `hideSnfgNodes` */
+function unhideSnfgNodes(plugin: PluginContext) {
+    const hiddenNodes = getExtensionState(plugin).hiddenNodes;
+    for (const nodeRef of Array.from(hiddenNodes.keys())) {
+        try {
+            setSubtreeVisibility(plugin.state.data, nodeRef, false); // false means visible
+        } catch {
+            // this is OK, the node has been removed
+        }
+    }
+    hiddenNodes.clear();
+}
```

Comment:
You are only hiding or showing all SFNG nodes, no need for the `getExtensionState` here at all.

Reply:
Add `entry: string` parameter to the hide function and use it to find the SFNG only in that subtree.

Reply:
Should be fixed now.